From 566a156edaf2e4340dd094d8d77a26c681b31286 Mon Sep 17 00:00:00 2001
From: Jean-Philippe Andre <jp.andre@samsung.com>
Date: Thu, 28 Apr 2016 16:05:00 +0900
Subject: [PATCH 42/79] Edje Edit: Fix compilation warning

---
 src/lib/edje/edje_edit.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/lib/edje/edje_edit.c b/src/lib/edje/edje_edit.c
index 02ae118..7739521 100644
--- a/src/lib/edje/edje_edit.c
+++ b/src/lib/edje/edje_edit.c
@@ -12934,7 +12934,7 @@ _edje_generate_source_state_proxy(Edje *ed, Edje_Part_Description_Common *pd, Ei
 {
    int attr_amount = 0;
    int indent_space = 0;
-   Eina_Bool ret;
+   Eina_Bool ret = 1;
    Edje_Part_Description_Proxy *pro;
 
    pro = (Edje_Part_Description_Proxy *)pd;
-- 
2.7.4

