From 05b9b86ea41771f7d187dccead01059a1c15c8a2 Mon Sep 17 00:00:00 2001
From: Vyacheslav Reutskiy <v.reutskiy@samsung.com>
Date: Thu, 16 Jun 2016 10:56:31 +0300
Subject: [PATCH 54/71] edje_edit: return NULL if part has not style attribute

---
 src/lib/edje/edje_edit.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/src/lib/edje/edje_edit.c b/src/lib/edje/edje_edit.c
index d45b5b4..026ea57 100644
--- a/src/lib/edje/edje_edit.c
+++ b/src/lib/edje/edje_edit.c
@@ -8120,7 +8120,10 @@ edje_edit_state_text_style_get(Evas_Object *obj, const char *part, const char *s
      return NULL;
 
    txt = (Edje_Part_Description_Text *)pd;
-   return eina_stringshare_add(txt->text.style.str);
+   if (txt->text.style.str)
+     return eina_stringshare_add(txt->text.style.str);
+   else
+     return NULL;
 }
 
 EAPI Eina_Bool
-- 
2.7.4

