From 3e8bd1562497f5f4ba7cbf9dc009af0db2930dec Mon Sep 17 00:00:00 2001
From: Andrii Kroitor <an.kroitor@samsung.com>
Date: Wed, 20 Jul 2016 17:36:26 +0300
Subject: [PATCH 64/71] edje_edit: comment out text and size classes

---
 src/lib/edje/edje_edit.c | 22 ++++++++++++++++++----
 1 file changed, 18 insertions(+), 4 deletions(-)

diff --git a/src/lib/edje/edje_edit.c b/src/lib/edje/edje_edit.c
index 35bd6ad..b3ff446 100644
--- a/src/lib/edje/edje_edit.c
+++ b/src/lib/edje/edje_edit.c
@@ -12222,6 +12222,7 @@ edje_edit_source_generate(Evas_Object *obj)
         BUF_APPEND(I0 "}\n\n");
      }
 
+   /*
    if (ed->file->text_classes)
      {
         BUF_APPEND(I0 "text_classes {\n");
@@ -12259,6 +12260,7 @@ edje_edit_source_generate(Evas_Object *obj)
 
         BUF_APPEND(I0 "}\n\n");
      }
+   */
 
    /* if images were found, print them */
    if (images || images_set)
@@ -12387,6 +12389,7 @@ edje_edit_source_generate(Evas_Object *obj)
 
 #undef COLLECT_RESOURCE
 
+/*
 static Eina_Bool
 _edje_generate_source_of_sizeclass(Edje *ed, const char *name, Eina_Strbuf *buf)
 {
@@ -12430,6 +12433,7 @@ _edje_generate_source_of_textclass(Edje *ed, const char *name, Eina_Strbuf *buf)
 
    return ret;
 }
+*/
 
 static Eina_Bool
 _edje_generate_source_of_colorclass(Edje *ed, const char *name, Eina_Strbuf *buf)
@@ -14135,10 +14139,12 @@ _edje_common_desc_diff_calculate(Edje_Part_Description_Common *ed,
                     (ed->color_class != NULL && inherit_pd->color_class != NULL &&
                      strcmp(ed->color_class, inherit_pd->color_class))) ? 1 : 0;
    diffs_amount += (ed->clip_to_id != inherit_pd->clip_to_id) ? 1 : 0;
+   /*
    diffs_amount += ((ed->size_class != NULL && inherit_pd->size_class == NULL) ||
                     (ed->size_class == NULL && inherit_pd->size_class != NULL) ||
                     (ed->size_class != NULL && inherit_pd->size_class != NULL &&
                      strcmp(ed->size_class, inherit_pd->size_class))) ? 1 : 0;
+   */
 
    /*Rel1 block*/
    diffs_amount += ((ed->rel1.relative_x != inherit_pd->rel1.relative_x) ||
@@ -14212,10 +14218,12 @@ _edje_text_desc_diff_calculate(Edje_Part_Description_Common *ed, Edje_Part_Descr
                     (ed_text->text.domain != NULL && inherit_pd_text->text.domain != NULL &&
                      strcmp(ed_text->text.domain, inherit_pd_text->text.domain))) ? 1 : 0;
 
+   /*
    diffs_amount += ((ed_text->text.text_class != NULL && inherit_pd_text->text.text_class == NULL) ||
                     (ed_text->text.text_class == NULL && inherit_pd_text->text.text_class != NULL) ||
                     (ed_text->text.text_class != NULL && inherit_pd_text->text.text_class != NULL &&
                      strcmp(ed_text->text.text_class, inherit_pd_text->text.text_class))) ? 1 : 0;
+   */
 
    diffs_amount += ((ed_text->text.font.str == NULL && inherit_pd_text->text.font.str != NULL) ||
                     (ed_text->text.font.str != NULL && inherit_pd_text->text.font.str == NULL) ||
@@ -14374,7 +14382,7 @@ _edje_generate_source_of_state(Evas_Object *obj, const char *part, const char *s
    Eina_Bool color_class = EINA_FALSE;
    Eina_Bool align = EINA_FALSE;
    Eina_Bool fixed = EINA_FALSE;
-   Eina_Bool size_class = EINA_FALSE;
+   /* Eina_Bool size_class = EINA_FALSE; */
    Eina_Bool minmul = EINA_FALSE;
    Eina_Bool min = EINA_FALSE, min_source = EINA_FALSE;
    Eina_Bool max = EINA_FALSE, max_source = EINA_FALSE;
@@ -14425,9 +14433,11 @@ _edje_generate_source_of_state(Evas_Object *obj, const char *part, const char *s
                   (pd->minmul.w == inherit_pd->minmul.w) &&
                   (pd->minmul.h == inherit_pd->minmul.h)) ? EINA_FALSE : EINA_TRUE;
 
+        /*
         size_class = ((pd->size_class == inherit_pd->size_class) ||
                       ((pd->size_class != NULL) && (inherit_pd->size_class != NULL) &&
                        (!strcmp(pd->size_class, inherit_pd->size_class)))) ? EINA_FALSE : EINA_TRUE;
+        */
 
         step = ((pd->step.x == inherit_pd->step.x) &&
                 (pd->step.y == inherit_pd->step.y)) ? EINA_FALSE : EINA_TRUE;
@@ -14468,7 +14478,7 @@ _edje_generate_source_of_state(Evas_Object *obj, const char *part, const char *s
         minmul = ((pd->minmul.have == 0) ||
                   ((pd->minmul.w == 1) && (pd->minmul.h == 1))) ? EINA_FALSE : EINA_TRUE;
 
-        size_class = (pd->size_class == NULL) ? EINA_FALSE : EINA_TRUE;
+        /* size_class = (pd->size_class == NULL) ? EINA_FALSE : EINA_TRUE; */
 
         step = ((pd->step.x == 0) && (pd->step.y == 0)) ? EINA_FALSE : EINA_TRUE;
 
@@ -14543,8 +14553,10 @@ _edje_generate_source_of_state(Evas_Object *obj, const char *part, const char *s
                                             TO_DOUBLE(pd->minmul.h),
                                             buf, &ret);
 
+   /*
    if (size_class)
      BUF_APPENDF(I5 "size_class: \"%s\";\n", pd->size_class);
+   */
 
    if (step)
      BUF_APPENDF(I5 "step: %d %d;\n", TO_INT(pd->step.x), TO_INT(pd->step.y));
@@ -15251,6 +15263,7 @@ _edje_generate_source(Evas_Object *obj)
           }
      }
    /* Size Classes */
+   /*
    if ((ll = edje_edit_size_classes_list_get(obj)))
      {
         BUF_APPEND(I0 "size_classes {\n");
@@ -15268,8 +15281,9 @@ _edje_generate_source(Evas_Object *obj)
              return NULL;
           }
      }
-
+   */
    /* Text Classes */
+   /*
    if ((ll = edje_edit_text_classes_list_get(obj)))
      {
         BUF_APPEND(I0 "text_classes {\n");
@@ -15287,7 +15301,7 @@ _edje_generate_source(Evas_Object *obj)
              return NULL;
           }
      }
-
+   */
    /* Color Classes */
    if ((ll = edje_edit_color_classes_list_get(obj)))
      {
-- 
2.7.4

