From f149a241100355f2f9d7a10c28eac4b516c7fbb0 Mon Sep 17 00:00:00 2001
From: Andrii Kroitor <an.kroitor@samsung.com>
Date: Wed, 20 Jul 2016 17:31:57 +0300
Subject: [PATCH 1/2] edje_edit: fix eo_parent_set calls

---
 src/lib/edje/edje_edit.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/lib/edje/edje_edit.c b/src/lib/edje/edje_edit.c
index c6df12f..35bd6ad 100644
--- a/src/lib/edje/edje_edit.c
+++ b/src/lib/edje/edje_edit.c
@@ -342,7 +342,7 @@ _edje_real_part_free(Edje *ed, Edje_Real_Part *rp)
    if ((rp->type == EDJE_RP_TYPE_SWALLOW) && (rp->typedata.swallow)
        && (rp->typedata.swallow->swallowed_object))
      {
-        eo_parent_set(rp->typedata.swallow->swallowed_object, evas_object_evas_get(ed->obj));
+        eo_do(rp->typedata.swallow->swallowed_object, eo_parent_set(evas_object_evas_get(ed->obj)));
         evas_object_smart_member_del(rp->typedata.swallow->swallowed_object);
         evas_object_event_callback_del(rp->typedata.swallow->swallowed_object,
                                        EVAS_CALLBACK_FREE, _edje_object_part_swallow_free_cb);
@@ -3173,7 +3173,7 @@ _edje_edit_real_part_add(Evas_Object *obj, const char *name, Edje_Part_Type type
         evas_object_layer_set(rp->object, evas_object_layer_get(ed->obj));
         if (ep->type == EDJE_PART_TYPE_SWALLOW)
           {
-             eo_parent_set(rp->object, ed->obj);
+             eo_do(rp->object, eo_parent_set(ed->obj));
           }
         else if (ep->type != EDJE_PART_TYPE_GROUP)
           {
-- 
2.7.4

